### PR TITLE
sparse: Add common files for Android 11 adaptations

### PR DESCRIPTION
**Pull request by mlehtima**

### CODE REVIEW

File: sparse-11/var/lib/environment/nemo/99-qtscenegraph.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This flag needs to be set, regardless of its value
+QT_OPENGL_NO_BGRA=1
```
Comment (krnlyng) [inline sparse-11/var/lib/environment/nemo/99-qtscenegraph.conf:2]: have you checked if this is still needed?

File: sparse-11/usr/libexec/droid-hybris/system/etc/init/disabled_services.rc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+service cameraserver cameraserver_HYBRIS_DISABLED
+
+service audioserver audioserver_HYBRIS_DISABLED
+
+service media mediaserver_HYBRIS_DISABLED
+
+service update_engine update_engine_HYBRIS_DISABLED
+
+service vendor.audio-hal-2-0 /vendor/bin/hw/android.hardware.audio@2.0-service_HYBRIS_DISABLED
```
Comment (krnlyng) [inline sparse-11/usr/libexec/droid-hybris/system/etc/init/disabled_services.rc:13]: Is it really audio hal 2.0?

File: sparse-11/usr/bin/droid/droid-hal-startup.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+cd /
+
+chmod 0755 /dev/binderfs
+ln -s /dev/binderfs/binder /dev/
+ln -s /dev/binderfs/hwbinder /dev/
+ln -s /dev/binderfs/vndbinder /dev/
+ln -s /dev/binderfs/puddlejumper /dev/
```
Comment (krnlyng) [inline sparse-11/usr/bin/droid/droid-hal-startup.sh:8]: We should remove these 3 from here.

File: sparse-11/usr/bin/droid/droid-hal-early-init.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+if ! grep -q hybris /system/etc/ld.config.29.txt; then
+    mount -o bind /usr/libexec/droid-hybris/system/etc/ld.config.29.txt /system/etc/ld.config.29.txt
```
Comment (krnlyng) [inline sparse-11/usr/bin/droid/droid-hal-early-init.sh:4]: Isn't it API level 30?

File: sparse-11/etc/gbinder.conf [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[General]
+ApiLevel = 29
```
Comment (krnlyng) [inline sparse-11/etc/gbinder.conf:2]: 30?

File: sparse-11/etc/ofono/ril_subscription.conf [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# Ofono configuration for Android 9
```
Comment (mkosola) [inline sparse-11/etc/ofono/ril_subscription.conf:1]: Should we fix Android 9 or adjust comment to some modular?

**Comment by Thaodan** [comment]
Do we have to have android version specific files in sparse/boot?

File: sparse-11/usr/bin/droid/droid-hal-early-init.sh (hunk 4 above)
Comment (mlehtima) [reply sparse-11/usr/bin/droid/droid-hal-early-init.sh:4]: This method doesn't work anymore, I will move the file.

File: sparse-11/etc/gbinder.conf (hunk 5 above)
Comment (mlehtima) [reply sparse-11/etc/gbinder.conf:2]: No support for api level 30 in gbinder yet.

File: sparse-11/usr/libexec/droid-hybris/system/etc/init/disabled_services.rc (hunk 2 above)
Comment (mlehtima) [reply sparse-11/usr/libexec/droid-hybris/system/etc/init/disabled_services.rc:13]: The service was renamed, wondering if this even should be here, this seems vendor specific, although probably in most cases the same.

File: sparse-11/etc/ofono/ril_subscription.conf (hunk 6 above)
Comment (mlehtima) [reply sparse-11/etc/ofono/ril_subscription.conf:1]: Will fix.

File: sparse-11/etc/gbinder.conf (hunk 5 above)
Comment (mlehtima) [reply sparse-11/etc/gbinder.conf:2]: It seems we can set 30 here even before the support is merged.

File: sparse-11/var/lib/environment/nemo/99-qtscenegraph.conf (hunk 1 above)
Comment (mlehtima) [reply sparse-11/var/lib/environment/nemo/99-qtscenegraph.conf:2]: Based on quick testing this wasn't needed but should be tested more before removing.

File: sparse-11/usr/bin/droid/droid-bootctl.sh [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# Check currentl used boot slot
+SLOT="$(/usr/libexec/droid-hybris/system/bin/bootctl get-current-slot)"
+
+# Check if current slot is marked successful
+/usr/libexec/droid-hybris/system/bin/bootctl is-slot-marked-successful $SLOT 2> /dev/null
```
Comment (krnlyng) [inline sparse-11/usr/bin/droid/droid-bootctl.sh:7]: Why do we need to check this before marking as successful?

Comment (mlehtima) [reply sparse-11/usr/bin/droid/droid-bootctl.sh:7]: Not sure why it was done like that, probably to stop it from always setting it if it's already ok.

**Review by krnlyng — APPROVED** [review]

**Pull request merged**